Here is some proof of the failure cases of `$ nix build .#nixosTests.scion-freestanding-deployment`

If `scion-pki certificate verify` fails:

```
...
scion01: must succeed: scion-pki certificate verify --trc /etc/scion/certs/*.trc /etc/scion/crypto/as/*.pem >&2
scion01 # Error: verification failed: chain did not verify against any selected TRC {errors=[verifying chain {trc_base=1; trc_serial=1}: x509: certificate has expired or is not yet valid: current time 2024-09-15T17:51:22Z is after 2024-09-13T20:35:35Z]}
scion01: output:
cleanup
kill machine (pid 10)
qemu-kvm: terminating on signal 15 from pid 7 (/nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3.12)
kill machine (pid 31)
qemu-kvm: terminating on signal 15 from pid 7 (/nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3.12)
kill machine (pid 52)
qemu-kvm: terminating on signal 15 from pid 7 (/nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3.12)
kill machine (pid 73)
qemu-kvm: terminating on signal 15 from pid 7 (/nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3.12)
kill machine (pid 95)
qemu-kvm: terminating on signal 15 from pid 7 (/nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3.12)
(finished: cleanup, in 0.05 seconds)
Traceback (most recent call last):
  File "/nix/store/07i3r7lw48irrnhgdv2ma25360z44a59-nixos-test-driver-1.1/bin/.nixos-test-driver-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/07i3r7lw48irrnhgdv2ma25360z44a59-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/__init__.py", line 146, in main
    driver.run_tests()
  File "/nix/store/07i3r7lw48irrnhgdv2ma25360z44a59-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/driver.py", line 166, in run_tests
    self.test_script()
  File "/nix/store/07i3r7lw48irrnhgdv2ma25360z44a59-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/driver.py", line 158, in test_script
    exec(self.tests, symbols, None)
  File "<string>", line 32, in <module>
  File "<string>", line 15, in succeed
  File "/nix/store/07i3r7lw48irrnhgdv2ma25360z44a59-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 611, in succeed
    raise Exception(f"command `{command}` failed (exit code {status})")
Exception: command `scion-pki certificate verify --trc /etc/scion/certs/*.trc /etc/scion/crypto/as/*.pem >&2` failed (exit code 1)
kill vlan (pid 8)
```

If `scion showpaths` and/or `scion ping` fails:

```
...
scion01: must succeed: /nix/store/79qyvacn46c8yspv3a04ir81jc1ig49b-ping-all-scion.sh >&2
scion02 # [   13.134379] scion-control[720]: 2024-09-15 17:39:56.564951+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "8b3977a1", "egress_interface": 1, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 1}}
scion02 # [   13.144923] scion-control[720]: 2024-09-15 17:39:56.575262+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "8b3977a1", "egress_interface": 2, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 2}}
scion02 # [   13.155361] scion-control[720]: 2024-09-15 17:39:56.575330+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "8b3977a1", "egress_interface": 3, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 3}}
scion01 # [   13.463350] scion-control[750]: 2024-09-15 17:39:56.830639+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "15bb516c", "egress_interface": 3, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 3}}
scion01 # [   13.474459] scion-control[750]: 2024-09-15 17:39:56.841988+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "15bb516c", "egress_interface": 1, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 1}}
scion01 # [   13.484938] scion-control[750]: 2024-09-15 17:39:56.842049+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "15bb516c", "egress_interface": 2, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 2}}
scion01 # Error: no path found
scion01 # Error: no path found
scion01 # Error: no path found
scion01 # Error: no path found
scion03 # [   17.879496] scion-control[723]: 2024-09-15 17:40:01.333539+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "f39c9dca", "egress_interface": 4, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 4}}
scion03 # [   17.889423] scion-control[723]: 2024-09-15 17:40:01.334057+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "f39c9dca", "egress_interface": 1, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 1}}
scion03 # [   17.899749] scion-control[723]: 2024-09-15 17:40:01.345241+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "f39c9dca", "egress_interface": 2, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 2}}
scion03 # [   17.910081] scion-control[723]: 2024-09-15 17:40:01.354687+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "f39c9dca", "egress_interface": 3, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 3}}
scion01 # Error: no path found
scion02 # [   18.632738] scion-control[720]: 2024-09-15 17:40:02.063004+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "8b3977a1", "egress_interface": 3, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 3}}
scion02 # [   18.642855] scion-control[720]: 2024-09-15 17:40:02.063196+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "8b3977a1", "egress_interface": 1, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 1}}
scion02 # [   18.652934] scion-control[720]: 2024-09-15 17:40:02.063242+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "8b3977a1", "egress_interface": 2, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 2}}
scion01 # [   18.954841] scion-control[750]: 2024-09-15 17:40:02.321691+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "15bb516c", "egress_interface": 3, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 3}}
scion01 # [   18.965887] scion-control[750]: 2024-09-15 17:40:02.332963+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "15bb516c", "egress_interface": 1, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 1}}
scion01 # [   18.976393] scion-control[750]: 2024-09-15 17:40:02.333339+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "15bb516c", "egress_interface": 2, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 2}}
scion01 # Error: no path found
scion01 # Error: no path found
scion01 # Error: no path found
scion03 # [   21.841499] scion-control[723]: 2024-09-15 17:40:05.296015+0000    INFO               control/main.go:522  Failed signer health check      {"err": {"msg": "failed to generate signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "github.com/scionproto/scion/control/trust.RenewingSigner.Sign github.com/scionproto/scion/control/trust/signer.go:44", "main.realMain.func4 github.com/scionproto/scion/control/cmd/control/main.go:521", "github.com/scionproto/scion/private/periodic.Func.Run github.com/scionproto/scion/private/periodic/periodic.go:85", "github.com/scionproto/scion/private/periodic.(*Runner).onTick github.com/scionproto/scion/private/periodic/periodic.go:213", "github.com/scionproto/scion/private/periodic.(*Runner).runLoop github.com/scionproto/scion/private/periodic/periodic.go:195", "github.com/scionproto/scion/private/periodic.StartWithMetrics.func1 github.com/scionproto/scion/private/periodic/periodic.go:145"]}}}
scion04 # [   22.002776] scion-control[727]: 2024-09-15 17:40:05.490604+0000    INFO               control/main.go:522  Failed signer health check      {"err": {"msg": "failed to generate signer", "cause": {"msg": "no certificate found", "stacktrace": ["github.com/scionproto/scion/private/trust.SignerGen.Generate github.com/scionproto/scion/private/trust/signer_gen.go:83", "github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:58", "github.com/scionproto/scion/control/trust.RenewingSigner.Sign github.com/scionproto/scion/control/trust/signer.go:44", "main.realMain.func4 github.com/scionproto/scion/control/cmd/control/main.go:521", "github.com/scionproto/scion/private/periodic.Func.Run github.com/scionproto/scion/private/periodic/periodic.go:85", "github.com/scionproto/scion/private/periodic.(*Runner).onTick github.com/scionproto/scion/private/periodic/periodic.go:213", "github.com/scionproto/scion/private/periodic.(*Runner).runLoop github.com/scionproto/scion/private/periodic/periodic.go:195", "github.com/scionproto/scion/private/periodic.StartWithMetrics.func1 github.com/scionproto/scion/private/periodic/periodic.go:145"], "num_private_keys": 1}}}
scion01 # Error: no path found
scion02 # [   22.589899] scion-control[720]: 2024-09-15 17:40:06.020298+0000    INFO               control/main.go:522  Failed signer health check      {"err": {"msg": "failed to generate signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "github.com/scionproto/scion/control/trust.RenewingSigner.Sign github.com/scionproto/scion/control/trust/signer.go:44", "main.realMain.func4 github.com/scionproto/scion/control/cmd/control/main.go:521", "github.com/scionproto/scion/private/periodic.Func.Run github.com/scionproto/scion/private/periodic/periodic.go:85", "github.com/scionproto/scion/private/periodic.(*Runner).onTick github.com/scionproto/scion/private/periodic/periodic.go:213", "github.com/scionproto/scion/private/periodic.(*Runner).runLoop github.com/scionproto/scion/private/periodic/periodic.go:195", "github.com/scionproto/scion/private/periodic.StartWithMetrics.func1 github.com/scionproto/scion/private/periodic/periodic.go:145"]}}}
scion05 # [   22.142275] scion-control[730]: 2024-09-15 17:40:06.605576+0000    INFO               control/main.go:522  Failed signer health check      {"err": {"msg": "failed to generate signer", "cause": {"msg": "no certificate found", "stacktrace": ["github.com/scionproto/scion/private/trust.SignerGen.Generate github.com/scionproto/scion/private/trust/signer_gen.go:83", "github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:58", "github.com/scionproto/scion/control/trust.RenewingSigner.Sign github.com/scionproto/scion/control/trust/signer.go:44", "main.realMain.func4 github.com/scionproto/scion/control/cmd/control/main.go:521", "github.com/scionproto/scion/private/periodic.Func.Run github.com/scionproto/scion/private/periodic/periodic.go:85", "github.com/scionproto/scion/private/periodic.(*Runner).onTick github.com/scionproto/scion/private/periodic/periodic.go:213", "github.com/scionproto/scion/private/periodic.(*Runner).runLoop github.com/scionproto/scion/private/periodic/periodic.go:195", "github.com/scionproto/scion/private/periodic.StartWithMetrics.func1 github.com/scionproto/scion/private/periodic/periodic.go:145"], "num_private_keys": 1}}}
scion01 # [   22.919453] scion-control[750]: 2024-09-15 17:40:06.286570+0000    INFO               control/main.go:522  Failed signer health check      {"err": {"msg": "failed to generate signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "github.com/scionproto/scion/control/trust.RenewingSigner.Sign github.com/scionproto/scion/control/trust/signer.go:44", "main.realMain.func4 github.com/scionproto/scion/control/cmd/control/main.go:521", "github.com/scionproto/scion/private/periodic.Func.Run github.com/scionproto/scion/private/periodic/periodic.go:85", "github.com/scionproto/scion/private/periodic.(*Runner).onTick github.com/scionproto/scion/private/periodic/periodic.go:213", "github.com/scionproto/scion/private/periodic.(*Runner).runLoop github.com/scionproto/scion/private/periodic/periodic.go:195", "github.com/scionproto/scion/private/periodic.StartWithMetrics.func1 github.com/scionproto/scion/private/periodic/periodic.go:145"]}}}
scion03 # [   22.879477] scion-control[723]: 2024-09-15 17:40:06.334225+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "f39c9dca", "egress_interface": 1, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 1}}
scion03 # [   22.889987] scion-control[723]: 2024-09-15 17:40:06.344643+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "f39c9dca", "egress_interface": 2, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 2}}
scion03 # [   22.900696] scion-control[723]: 2024-09-15 17:40:06.344921+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "f39c9dca", "egress_interface": 3, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 3}}
scion03 # [   22.912110] scion-control[723]: 2024-09-15 17:40:06.344978+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "f39c9dca", "egress_interface": 4, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 4}}
scion01 # Error: no path found
scion02 # [   23.633221] scion-control[720]: 2024-09-15 17:40:07.063391+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "8b3977a1", "egress_interface": 3, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 3}}
scion02 # [   23.643401] scion-control[720]: 2024-09-15 17:40:07.063602+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "8b3977a1", "egress_interface": 1, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 1}}
scion02 # [   23.653572] scion-control[720]: 2024-09-15 17:40:07.063688+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "8b3977a1", "egress_interface": 2, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 2}}
scion01 # [   23.954925] scion-control[750]: 2024-09-15 17:40:07.321761+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "15bb516c", "egress_interface": 3, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 3}}
scion01 # [   23.965707] scion-control[750]: 2024-09-15 17:40:07.321929+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "15bb516c", "egress_interface": 1, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 1}}
scion01 # [   23.975910] scion-control[750]: 2024-09-15 17:40:07.322050+0000    INFO       beaconing/originator.go:118  Unable to originate on interface        {"debug_id": "15bb516c", "egress_interface": 2, "err": {"msg": "creating beacon", "cause": {"msg": "extending segment", "cause": {"msg": "getting signer", "cause": {"msg": "no signer cached, reload interval has not passed", "stacktrace": ["github.com/scionproto/scion/control/trust.(*CachingSignerGen).Generate github.com/scionproto/scion/control/trust/signer_gen.go:53", "main.realMain.func18 github.com/scionproto/scion/control/cmd/control/main.go:782", "github.com/scionproto/scion/control/beaconing.SignerGenFunc.Generate github.com/scionproto/scion/control/beaconing/extender.go:52", "github.com/scionproto/scion/control/beaconing.(*DefaultExtender).Extend github.com/scionproto/scion/control/beaconing/extender.go:114", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).createBeacon github.com/scionproto/scion/control/beaconing/originator.go:208", "github.com/scionproto/scion/control/beaconing.(*beaconOriginator).originateBeacon github.com/scionproto/scion/control/beaconing/originator.go:163", "github.com/scionproto/scion/control/beaconing.(*Originator).originateBeacons.func2 github.com/scionproto/scion/control/beaconing/originator.go:117"]}}}, "egress_interface": 2}}
scion01 # Error: no path found
scion01: output:
cleanup
kill machine (pid 10)
qemu-kvm: terminating on signal 15 from pid 7 (/nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3.12)
kill machine (pid 31)
qemu-kvm: terminating on signal 15 from pid 7 (/nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3.12)
kill machine (pid 52)
qemu-kvm: terminating on signal 15 from pid 7 (/nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3.12)
kill machine (pid 73)
qemu-kvm: terminating on signal 15 from pid 7 (/nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3.12)
kill machine (pid 94)
qemu-kvm: terminating on signal 15 from pid 7 (/nix/store/h3i0acpmr8mrjx07519xxmidv8mpax4y-python3-3.12.5/bin/python3.12)
(finished: cleanup, in 0.04 seconds)
Traceback (most recent call last):
  File "/nix/store/07i3r7lw48irrnhgdv2ma25360z44a59-nixos-test-driver-1.1/bin/.nixos-test-driver-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/07i3r7lw48irrnhgdv2ma25360z44a59-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/__init__.py", line 146, in main
    driver.run_tests()
  File "/nix/store/07i3r7lw48irrnhgdv2ma25360z44a59-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/driver.py", line 166, in run_tests
    self.test_script()
  File "/nix/store/07i3r7lw48irrnhgdv2ma25360z44a59-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/driver.py", line 158, in test_script
    exec(self.tests, symbols, None)
  File "<string>", line 35, in <module>
  File "<string>", line 15, in succeed
  File "/nix/store/07i3r7lw48irrnhgdv2ma25360z44a59-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 611, in succeed
    raise Exception(f"command `{command}` failed (exit code {status})")
Exception: command `/nix/store/79qyvacn46c8yspv3a04ir81jc1ig49b-ping-all-scion.sh >&2` failed (exit code 1)
kill vlan (pid 8)
```

--- a/nixos/tests/scion/freestanding-deployment/default.nix
+++ b/nixos/tests/scion/freestanding-deployment/default.nix
@@ -131,25 +131,35 @@ in
       addresses="42-ffaa:1:1 42-ffaa:1:2 42-ffaa:1:3 42-ffaa:1:4 42-ffaa:1:5"
       timeout=100
       wait_for_all() {
+        ret=0
         for as in "$@"
         do
           scion showpaths $as --no-probe > /dev/null
-          return 1
+          ret=$?
+          if [ "$ret" -ne "0" ]; then
+            break
+          fi
         done
-        return 0
+        return $ret
       }
       ping_all() {
+        ret=0
         for as in "$@"
         do
           scion ping "$as,127.0.0.1" -c 3
+          ret=$?
+          if [ "$ret" -ne "0" ]; then
+            break
+          fi
         done
-        return 0
+        return $ret
       }
       for i in $(seq 0 $timeout); do
-        wait_for_all $addresses && exit 0
-        ping_all $addresses && exit 0
         sleep 1
+        wait_for_all $addresses || continue
+        ping_all $addresses && exit 0
       done
+      exit 1
     '';
   in
   ''

--- a/nixos/tests/scion/freestanding-deployment/default.nix
+++ b/nixos/tests/scion/freestanding-deployment/default.nix
@@ -193,6 +193,9 @@ in
     # Wait for scion-control.service on all instances
     wait_for_unit("scion-control.service")
 
+    # Ensure cert is valid against TRC
+    succeed("scion-pki certificate verify --trc /etc/scion/certs/*.trc /etc/scion/crypto/as/*.pem >&2")
+
     # Execute pingAll command on all instances
     succeed("${pingAll} >&2")
 

